### PR TITLE
Display value for player

### PR DESCRIPTION
**Pull request by benbarn313**
https://github.com/holoviz/panel/assets/55669975/0ea9535c-1dbf-417a-b31e-42b0f7fec94d

PR closes issue https://github.com/holoviz/panel/issues/229

Added display value for the Player widget
Added the param "value_location" to the Slider, which sets where the value is displayed

Planning on adding support for the Name param as well

### CODE REVIEW

File: Untitled.ipynb [hunk 1]
```diff
@@ -0,0 +1,135 @@
+{
```
Comment (ahuang11) [inline Untitled.ipynb:1]: Oops, seems like this was accidentally committed.

File: panel/widgets/player.py [hunk 2]
```diff
@@ -21,6 +21,10 @@
 
 class PlayerBase(Widget):
 
+
+    #value_location = param.String(default="", allow_None=True, doc="""
```
Comment (ahuang11) [inline panel/widgets/player.py:25]:
```suggestion
```

File: panel/widgets/player.py [hunk 3]
```diff
@@ -41,6 +45,10 @@ class PlayerBase(Widget):
 
     height = param.Integer(default=80)
 
+    value_location = param.String(default="top_center", allow_None=True, doc="""
```
Comment (ahuang11) [inline panel/widgets/player.py:48]: Maybe ObjectSelector would be a better choice here

File: panel/models/widgets.py [hunk 4]
```diff
@@ -21,6 +21,7 @@ class Player(Widget):
     """
     The Player widget provides controls to play through a number of frames.
     """
+    #name = String("", help="""Name to display above the widget""")
```
Comment (ahuang11) [inline panel/models/widgets.py:24]:
```suggestion
```

File: panel/models/player.ts [hunk 5]
```diff
@@ -437,14 +473,16 @@ export class Player extends Widget {
   static {
     this.prototype.default_view = PlayerView
 
-    this.define<Player.Props>(({ Boolean, Int }) => ({
+    this.define<Player.Props>(({ Boolean, Int, String }) => ({
+      //name: [String, ""],
```
Comment (ahuang11) [inline panel/models/player.ts:477]:
```suggestion
```

File: panel/models/player.ts [hunk 6]
```diff
@@ -409,13 +443,15 @@ export const LoopPolicy = Enum("once", "loop", "reflect")
 export namespace Player {
   export type Attrs = p.AttrsOf<Props>
   export type Props = Widget.Props & {
+    //name: p.Property<string>
```
Comment (ahuang11) [inline panel/models/player.ts:446]:
```suggestion
```

**Comment by benbarn313** [comment]
@philippjfr can you take a look at this?

File: panel/models/widgets.py [hunk 7]
```diff
@@ -30,6 +33,9 @@ class Player(Widget):
 
     value_throttled = Int(0, help="Current throttled value of the player app")
 
+    value_location = String("top_center", help="""Location to display
```
Comment (ahuang11) [inline panel/models/widgets.py:36]:
```suggestion
    value_location = String("top_left", help="""Location to display
```

File: panel/widgets/player.py [hunk 8]
```diff
@@ -78,7 +89,7 @@ class Player(PlayerBase):
 
     :Example:
 
-    >>> Player(name='Player', start=0, end=100, value=32, loop_policy='loop')
+    >>> Player(name='Player', start=0, end=100, value=32, loop_policy='loop', value_location = 'top_center')
```
Comment (ahuang11) [inline panel/widgets/player.py:92]:
```suggestion
    >>> Player(name='Player', start=0, end=100, value=32, loop_policy='loop', value_location='top_center')
```

File: panel/widgets/player.py [hunk 9]
```diff
@@ -132,7 +143,8 @@ class DiscretePlayer(PlayerBase, SelectBase):
     >>> DiscretePlayer(
     ...     name='Discrete Player',
     ...     options=[2, 4, 8, 16, 32, 64, 128], value=32,
-    ...     loop_policy='loop'
+    ...     loop_policy='loop',
+    ...     value_location = 'top_center'
```
Comment (ahuang11) [inline panel/widgets/player.py:147]:
```suggestion
    ...     value_location='top_left'
```

File: panel/widgets/player.py [hunk 10]
```diff
@@ -36,19 +37,29 @@ class PlayerBase(Widget):
...
     step = param.Integer(default=1, doc="""
         Number of frames to step forward and back by on each event.""")
 
     height = param.Integer(default=80)
 
+    value_location = param.ObjectSelector(
+        objects=["top_left", "top_center", "top_right"], doc="""
+        Location to display the value of the slider
+        ("top_left", "top_center", "top_right")""")
+
```
Comment (ahuang11) [inline panel/widgets/player.py:52]: We need to add these new params to the reference docs params intro, and maybe a brief example in one of the cells:

`panel/examples/reference/widgets` for Player.ipynb and DiscretePlayer.ipynb I think.

See https://github.com/holoviz/panel/pull/6087/files for an example

**Review by ahuang11 — COMMENTED** [review]
Awesome! You're becoming the Player* expert!

Lots of people will appreciate this feature :)

**Comment by philippjfr** [comment]
This is looking great! In practice we mostly use `DiscretePlayer` and there we don't currently have the corresponding labels available in the frontend right? So I think we have to add (optional) support for sending a list of string labels which will be displayed instead of the Player index.

File: panel/models/player.ts [hunk 11]
```diff
@@ -79,6 +81,8 @@ export class PlayerView extends WidgetView {
 
   connect_signals(): void {
     super.connect_signals()
+    this.connect(this.model.properties.name.change, () => this.update_title())
```
Comment (philippjfr) [inline panel/models/player.ts:84]:
```suggestion
    this.connect(this.model.properties.title.change, () => this.update_title())
```

**Comment by ahuang11** [comment]
Superseded by https://github.com/holoviz/panel/pull/7060

**Pull request closed without merging**